### PR TITLE
Add some basic logging for AuthN/AuthZ

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/aspnet/Security/issues/491

@Tratcher @Eilon @divega 

Only added the important top level logging needed for the review later this week

### CODE REVIEW

File: src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs [hunk 1]
```diff
@@ -240,6 +243,7 @@ protected Task<AuthenticateResult> HandleAuthenticateOnceAsync()
             if (ShouldHandleScheme(context.AuthenticationScheme, handleAutomatic: false))
             {
                 SignInAccepted = true;
+                Logger.LogInformation(3, $"AuthenticationScheme: {Options.AuthenticationScheme} signed in.");
                 await HandleSignInAsync(context);
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs]: It's better to log afterwards in case there are exceptions.

**Comment by HaoK** [comment]
Updated moving logging after method calls

**Comment by divega** [comment]
LGTM. CC @rynowak and @loudej who I think can look at this from a wider logging perspective.

File: src/Microsoft.AspNet.Authorization/DefaultAuthorizationService.cs [hunk 2]
```diff
@@ -33,7 +50,17 @@ public DefaultAuthorizationService(IOptions<AuthorizationOptions> options, IEnum
...
-            return authContext.HasSucceeded;
+
+            if (authContext.HasSucceeded)
+            {
+                _logger.LogInformation(0, $"Authorization was successful for user: {user?.Identity?.Name}.");
+                return true;
+            }
+            else
+            {
+                _logger.LogInformation(0, $"Authorization failed for user: {user?.Identity?.Name}.");
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Authorization/DefaultAuthorizationService.cs]: 0?

Comment (HaoK) [inline src/Microsoft.AspNet.Authorization/DefaultAuthorizationService.cs]: Grrr, copy paste fail, yeah event ids, manually assigning them for now, 0 and 1 :)

**Comment by Tratcher** [comment]
:shipit:

File: src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs [hunk 3]
```diff
@@ -104,6 +104,7 @@ protected string CurrentUri
                 if (ticket?.Principal != null)
                 {
                     Context.User = SecurityHelper.MergeUserPrincipal(Context.User, ticket.Principal);
+                    Logger.LogInformation(0, $"HttContext.User merged via AutomaticAuthentication from authenticationScheme: {Options.AuthenticationScheme}.");
```
Comment (Eilon) [inline src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs]: You know, come to think of it, I think these aren't supposed to use string interpolation or format strings.

Logging has its own format stuff for structured data, something like:

```c#
Logger.LogInformation(0, "HttContext.User merged via AutomaticAuthentication from authenticationScheme: {AuthenticationScheme}.", Options.AuthenticationScheme);
```

Note that there's no string formatting going on here, but Logging will see the `{foo}` in the string and create an associated structured data element with `AuthenticationScheme = Options.AuthenticationScheme`.

I missed this in my earlier logging change.

Comment (HaoK) [inline src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs]: Ah that's right, removed all interpolation

**Comment by HaoK** [comment]
e0464c950853909aa3491c65b519db47bb22679f

**Pull request closed without merging**